src/rgw/rgw_rados.cc

```diff
@@ -2279,38 +2281,87 @@ int RGWPutObjProcessor_Atomic::handle_data(bufferlist& bl, off_t ofs, MD5 *hash,
   *again = false;
 
   *phandle = NULL;
+
+  bufferlist in_bl;
+
+  // compression stuff
+  bool compression_enabled = store->ctx()->_conf->rgw_compression_type != "none";
+  if ((ofs > 0 && compressed) ||                                // if previous part was compressed
+      (ofs == 0 && compression_enabled)) {   // or it's the first part and flag is set
+    ldout(store->ctx(), 10) << "Compression for rgw is enabled, compress part " << bl.length() << dendl;
+    CompressorRef compressor = Compressor::create(store->ctx(), store->ctx()->_conf->rgw_compression_type);
```

Comment:
@ifed01 It's cached in plugin, no need to store it twice, for example
https://github.com/ceph/ceph/blob/master/src/compressor/zlib/CompressionPluginZlib.cc#L34